account_payment_term_partner_holiday/models/account_payment_term.py

```diff
@@ -2,24 +2,43 @@
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
 
 from odoo import api, fields, models
+from dateutil.relativedelta import relativedelta
 
 
 class AccountPaymentTerm(models.Model):
     _inherit = "account.payment.term"
 
     @api.one
     def compute(self, value, date_ref=False):
+        """If date_to_check has changed is because initial date is in holiday
+        period and these change make necessary to re-compute again to be sure
+        that date returnes is according to payment term (+10 days or custom
+        payment days).
+        """
         result = super().compute(value=value, date_ref=date_ref)[0]
         ctx = self.env.context
         partner_id = ctx.get("move_partner_id", ctx.get("default_partner_id"))
         if partner_id:
             partner = self.env["res.partner"].browse(partner_id)
             result2 = []
             for item in result:
-                new_date_item = partner._get_valid_due_date(item[0])
-                if new_date_item != item[0]:
-                    result2.append((fields.Date.to_string(new_date_item), item[1]))
-                else:
-                    result2.append(item)
+                new_item = (item[0], item[1])
+                date_to_check = item[0]
+                res_is_date_in_holiday = partner.is_date_in_holiday(date_to_check)
+                while res_is_date_in_holiday:
+                    date_to_check = res_is_date_in_holiday[1] + relativedelta(days=1)
+                    date_to_check = fields.Date.to_string(date_to_check)
+                    res = super().compute(
+                        value=value, date_ref=date_to_check
+                    )[0]
+                    res_is_date_in_holiday = partner.is_date_in_holiday(res[0][0])
+                # Need to re-compute according to initial comment
+                if date_to_check != item[0]:
+                    res = super().compute(
+                        value=value, date_ref=date_to_check
+                    )[0]
+                    date_to_check = res[0][0]
+                new_item = (date_to_check, item[1])
+                result2.append(new_item)
```

Comment:
If this is the actual desired behavior (always recompute the date), I think this can be simplified, following the current logic:

```suggestion
                new_date_item = partner._get_valid_due_date(item[0])
                if new_date_item != item[0]:
                    # Re-compute
                    res = super().compute(
                        value=value, date_ref=new_date_item
                    )[0]
                    result2.append(res[0])
                else:
                    result2.append(item)
            result = result2
```

This way, you avoid:

1. Repeating code that you already have in `res.partner` for the incremental date check
2. Re-compute the payment term result date twice (as for your explanation, you just need to recompute once when you already have the first available date computed)